### PR TITLE
Plug together test framework

### LINKED ISSUE #5380: Combine integration test framework interfaces and functor over tests to reduce duplication

### PR DESCRIPTION
**Pull request by nholland94**
I took the work that @deepthiskumar and I did on the infrastructure side of the integration test framework and refactored it into an abstract cloud engine for the integration test framework. I used this setup to build an easier functor interface for defining tests, and moved the block production test into that setup. Along the way, I realized that we needed the ability to wait for initialization of nodes, so I added support for that to the stack driver log engine. This is able to deploy networks via coda-automation right now, but the networks have some issues and the block producer is currently crashing. I will continue to work on fixing this on the side, but am putting this PR up so that we can merge the intermediate work in the mean time.

Closes #5380.

### CODE REVIEW

File: src/app/test_executive/test_executive.ml [hunk 1]
```diff
@@ -1,50 +1,62 @@
...
-  ; proof_level= Runtime_config.Proof_keys.Level.Full
-  ; txpool_max_size= 3000
-  ; block_producers=
-      [{balance= bal "7500"}; {balance= bal "5000"}; {balance= bal "2500"}]
-  ; num_snark_workers= 4
-  ; snark_worker_fee= "0.025"
-  ; snark_worker_public_key=
-      "4vsRCVQZ41uqXfVVfkBNUuNNS7PgSJGdMDNAyKGDdU1WkdxxyxQ7oMdFcjDRf45fiGKkdYKkLPBrE1KnxmyBuvaTW97A5C8XjNSiJmvo9oHa4AwyVsZ3ACaspgQ3EyxQXk6uujaxzvQhbLDx"
-  }
+type inputs = {test: test; coda_image: string}
```
Comment (psteckler) [inline src/app/test_executive/test_executive.ml:8]: should `Test_functor` be `Test_functor_S` or such to indicate it's a signature?

Comment (nholland94) [reply src/app/test_executive/test_executive.ml:8]: `Test_functor_S` seems weird to me. Perhaps `Test_functor_intf` fits our current pattern better?

Comment (psteckler) [reply src/app/test_executive/test_executive.ml:8]: sure

File: src/lib/integration_test_cloud_engine/coda_automation.ml [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+        ; txpool_max_size
+        ; block_producers
+        ; num_snark_workers
+        ; snark_worker_fee
+        ; snark_worker_public_key } =
+      test_config
+    in
+    let testnet_name = "integration-test-" ^ test_name in
+    (* HARD CODED NETWORK VALUES *)
+    let coda_automation_location = "../automation" in
```
Comment (ghost-not-in-the-shell) [inline src/lib/integration_test_cloud_engine/coda_automation.ml:79]: Currently the path to coda-automation is hard coded.

**Pull request merged**